src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -1761,7 +1812,10 @@ static int do_map(int argc, const char *argv[], Config *cfg, bool reconnect)
     if (r < 0)
       goto close_nbd;
 
-    cout << cfg->devpath << std::endl;
+    if (use_netlink && cfg->show_cookie && !cfg->cookie.empty())
+      cout << cfg->devpath << " " << get_cookie(cfg->devpath) << std::endl;
```

Comment:
Might be good to `ceph_assert` that `get_cookie()` returns exactly `cfg->cookie`.

Reply:
Sure.

Reply:
Isn't that done only on attach?  The assert I was envisioning would ensure that the cookie that we generate on map and pass to the kernel is returned through `get_cookie()` intact.

Reply:
Right, but that was intentional.
Having an assertion here will make us future-proof :+1: